### PR TITLE
Single Command line worker deploy/update with ansible

### PR DESCRIPTION
**Pull request by vepkenez**
Needs people to test it.
Does it work (works for @vepkenez and @mswilkison on MacOS).
Does it need more configurability?
Are there more deployment environments that should have examples?

See the [README](https://github.com/nucypher/nucypher/blob/a331dd153fbc8bd43926451fa7611ee3764816ee/deploy/ansible/worker/README.md)

### CODE REVIEW

**Comment by KPrasch** [comment]
Looking Great - Still reviewing and testing. :-)

File: deploy/ansible/worker/include/run_nucypher_ursula.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        name: ursula
+        state: started
+        image: "{{ nucypher_image | default('nucypher/nucypher:latest') }}"
+        restart_policy: "unless-stopped"
+        command: "nucypher ursula run --teacher discover.nucypher.network:9151 --poa {{nucypher_ursula_run_options | default('')}}"
+        volumes:
+          - /home/nucypher:/root/.local/share/
+        ports:
+          - "9151:9151"
+          - "9555:9555"
```
[the remark this answers is not in the document]
Comment (vepkenez) [reply deploy/ansible/worker/include/run_nucypher_ursula.yml]: that is the port for prometheus

[the remark this answers is not in the document]
Comment (KPrasch) [reply deploy/ansible/worker/include/run_nucypher_ursula.yml]: This should be disabled by default.

File: deploy/ansible/worker/include/init_nucypher.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      become: yes
+      become_user: nucypher
+      stat:
+        path: /home/nucypher/nucypher/ursula.json
+      register: ursula_check
+
+    - name: Find my public ip
+      when: ursula_check.stat.exists == False
+      uri:
+        url: http://ifconfig.me/ip
```
Comment (jMyles) [inline deploy/ansible/worker/include/init_nucypher.yml]: Ladies and gentlemen, the internet.

File: deploy/ansible/worker/inventory.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            geth_dir: '/home/nucypher/geth/.ethereum/goerli/'
+            geth_container_datadir: "/root/.ethereum/goerli"
+            etherscan_domain: goerli.etherscan.io
+
+            # these can be overridden at the instance level if desired
+            NUCYPHER_KEYRING_PASSWORD: xxxxxxxxxxxxxxxxxxxxxxxpanda
+            NUCYPHER_WORKER_ETH_PASSWORD: yyyyyyyyyyyyyyyyyyyystainpants
+            #nucypher_ursula_run_options: "--debug"
+            #nucypher_ursula_init_options: "--debug"
+            # nucypher_image: "nucypher/nucypher:v2.0.0-beta.0"
```
Comment (vepkenez) [inline deploy/ansible/worker/inventory.yml]: replace this line with:
`nucypher_image: nucypher/nucypher:experimental`
That is a version of master with some fixes in it as well as a json status page.

File: deploy/ansible/worker/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#### single command line worker UPDATE
+
+updates all your existing nodes to the latest nucypher docker image
+
+1. from `/nucypher/deploy/ansible/` run `ansible-playbook worker/update_remote_workers.yml -i worker/inventory.yml -l cassandra`
+
+
+#### other commands to try 
+
+* `ansible-playbook worker/get_workers_status.yml -i ~/Documents/my_inventory.yml -l cassandra`
```
Comment (vepkenez) [inline deploy/ansible/worker/README.md]: @mswilkison try this as well.

```
"msg": [
        "MediumTurquoise Clock Purple Heart",
        "staker address:        0xB1171EE4fd7E54ed960ccF9ECa383E0f3B1C6B06",
        "worker address:        0x94fBC43fC56d7D3A3002E85C3575009316BcCcB9",
        "rest url:              https://52.34.68.134:9151",
        "missing_confirmations: 0",
        "last active period:    18294",
        "balances:",
        "   ETH:                0",
        "   NU:                 0"
    ]
```

**Comment by KPrasch** [comment]
Depends on #1641

File: deploy/ansible/worker/README.md (hunk 4 above)
Comment (KPrasch) [reply deploy/ansible/worker/README.md]: (from #1641 )

File: deploy/ansible/worker/inventory.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          vars:
+            network_name: cassandra
+            geth_options: "--goerli"
+            geth_dir: '/home/nucypher/geth/.ethereum/goerli/'
+            geth_container_datadir: "/root/.ethereum/goerli"
+            etherscan_domain: goerli.etherscan.io
+
+            # these can be overridden at the instance level if desired
+            NUCYPHER_KEYRING_PASSWORD: xxxxxxxxxxxxxxxxxxxxxxxpanda
+            NUCYPHER_WORKER_ETH_PASSWORD: yyyyyyyyyyyyyyyyyyyystainpants
```
Comment (KPrasch) [inline deploy/ansible/worker/inventory.yml]: Beyond the scope of  *this* PR - but note that it is possible to use inline encrypted ansible vault strings here, if the user is familiar with ansible tooling.

File: deploy/ansible/worker/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+#### single command line worker deployment
+
```
Comment (KPrasch) [inline deploy/ansible/worker/README.md]: I'd like to see this move into a formalized guide in the official documentation.

File: deploy/ansible/worker/include/run_geth.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
+- name: "Sync/Run Geth"
```
Comment (KPrasch) [inline deploy/ansible/worker/include/run_geth.yml]: On mainnet, this play will need to stand alone for a multi-day syncing time.

Comment (mswilkison) [reply deploy/ansible/worker/include/run_geth.yml]: Maybe we add a flag or something in inventory.yml to allow for light clients?

File: deploy/ansible/worker/inventory.yml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+          children: # add or remove children as needed
+
+            #### digital ocean uses default user "root" so we have to group these separately
+            digitalocean: # this can be anything... "my_nodes" or "home"
+              vars:
+                default_user: "root"
+              hosts:
+                192.168.1.1:
+                  WORKER_ACCT_KEYSTORE_PATH: "/Users/macperson/Library/Ethereum/goerli/keystore/UTC--2020-01-21T02-15-18.405738000Z--02e8cbf55e781ad4ca331fe5274be93814d760d0"
+                  NUCYPHER_WORKER_ADDRESS: "0x02e8cbf55E781AD4cA331fe5274Be93814D760D0"
```
Comment (mswilkison) [inline deploy/ansible/worker/inventory.yml]: Maybe we comment these out by default?

File: deploy/ansible/worker/inventory.yml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        cassandra:
+          children: # add or remove children as needed
+
+            #### digital ocean uses default user "root" so we have to group these separately
+            digitalocean: # this can be anything... "my_nodes" or "home"
+              vars:
+                default_user: "root"
+              hosts:
+                192.168.1.1:
+                  WORKER_ACCT_KEYSTORE_PATH: "/Users/macperson/Library/Ethereum/goerli/keystore/UTC--2020-01-21T02-15-18.405738000Z--02e8cbf55e781ad4ca331fe5274be93814d760d0"
```
Comment (mswilkison) [inline deploy/ansible/worker/inventory.yml]:
```suggestion
                  ### By default, the worker account is auto-generated on the remote server.
                  ### Alternatively, you can generate it locally and copy it over.
                  # WORKER_ACCT_KEYSTORE_PATH: "/Users/macperson/Library/Ethereum/goerli/keystore/UTC--2020-01-21T02-15-18.405738000Z--02e8cbf55e781ad4ca331fe5274be93814d760d0"
```

File: deploy/ansible/worker/inventory.yml (hunk 8 above)
Comment (mswilkison) [inline deploy/ansible/worker/inventory.yml]:
```suggestion
                  # NUCYPHER_WORKER_ADDRESS: "0x02e8cbf55E781AD4cA331fe5274Be93814D760D0"
```

File: deploy/ansible/worker/inventory.yml [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+            #### amazon configures new instances with a .pem keypair based auth
+            # so we need this ansible_ssh_private_key_file variable
+            amazon:
+              vars:
+                ansible_ssh_private_key_file: ~/Downloads/damon-ansible-testing.pem
+                default_user: "ubuntu"  # or root for digital ocean
+              hosts:
+                # add a host for each worker/staker
+                cassandra1.mystakednodez.com:
+                  WORKER_ACCT_KEYSTORE_PATH: "/home/ubuntu/.ethereum/goerli/keystore/UTC--2020-01-21T02-15-33.342507000Z--d9e7eC6f9bB558c739CDdbAD5c38F170F1571077"
```
Comment (mswilkison) [inline deploy/ansible/worker/inventory.yml]:
```suggestion
                  ### By default, the worker account is auto-generated on the remote server.
                  ### Alternatively, you can generate it locally and copy it over.
                  # WORKER_ACCT_KEYSTORE_PATH: "/home/ubuntu/.ethereum/goerli/keystore/UTC--2020-01-21T02-15-33.342507000Z--d9e7eC6f9bB558c739CDdbAD5c38F170F1571077"
```

File: deploy/ansible/worker/inventory.yml [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+                  NUCYPHER_WORKER_ADDRESS: "0xd9e7eC6f9bB558c739CDdbAD5c38F170F1571077"
```
Comment (mswilkison) [inline deploy/ansible/worker/inventory.yml]:
```suggestion
                  # NUCYPHER_WORKER_ADDRESS: "0xd9e7eC6f9bB558c739CDdbAD5c38F170F1571077"
```

File: deploy/ansible/worker/inventory.yml [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+                  NUCYPHER_STAKER_ADDRESS: "0x4ffaEAe86c6A6f2C810F048877fbe1FBB2B27606"
+                86.75.30.9:
+                  WORKER_ACCT_KEYSTORE_PATH: "/home/ubuntu/.ethereum/goerli/keystore/UTC--2020-01-21T02-15-33.342507000Z--d9e7eC6fddde58c739CDdbAD5c38F170F1571077"
```
Comment (mswilkison) [inline deploy/ansible/worker/inventory.yml]:
```suggestion
                  # WORKER_ACCT_KEYSTORE_PATH: "/home/ubuntu/.ethereum/goerli/keystore/UTC--2020-01-21T02-15-33.342507000Z--d9e7eC6fddde58c739CDdbAD5c38F170F1571077"
```

File: deploy/ansible/worker/inventory.yml [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+                  NUCYPHER_WORKER_ADDRESS: "0xd9e7eC6fddde58c739CDdbAD5c38F170F1571077"
```
Comment (mswilkison) [inline deploy/ansible/worker/inventory.yml]:
```suggestion
                  # NUCYPHER_WORKER_ADDRESS: "0xd9e7eC6fddde58c739CDdbAD5c38F170F1571077"
```

File: deploy/ansible/worker/include/run_geth.yml (hunk 7 above)
Comment (vepkenez) [reply deploy/ansible/worker/include/run_geth.yml]: @mswilkison this is available already through the geth_options envvar in the inventory.  It defaults to`—goerli` for cassandra

**Pull request merged**